internal/webhooks/clusterclass_test.go

```diff
@@ -1318,6 +1320,854 @@ func TestClusterClassValidationWithClusterAwareChecks(t *testing.T) {
 	}
 }
 
+func TestClusterClassValidationWithVariableChecks(t *testing.T) {
+	// NOTE: ClusterTopology feature flag is disabled by default, thus preventing to create or update ClusterClasses.
+	// Enabling the feature flag temporarily for this test.
+	defer utilfeature.SetFeatureGateDuringTest(t, feature.Gates, feature.ClusterTopology, true)()
+
+	clusterClassBuilder := builder.ClusterClass(metav1.NamespaceDefault, "class1").
```

Comment:
reusing the clusterClassBuilder only works because we always overwrite the same field, right? (otherwise we would collect fields from one test case to another).

WDYT about the alternative of just using arrays of ClusterClassVariables in the table and then building the full ClusterClass in the for loop? (should also safe as ~4 lines per test case, but that's not my primary reason :))

Reply:
I'm going to do a full review / rewrite of all the validation test cases, there's a lot of repetition and redundancy due to the piece by piece nature of the implementation. I'll push this to a future PR.